### PR TITLE
[pytorch][build] Unpin Pillow and upgrade to Pillow 8.x on PT 1.7 dockerfiles

### PR DESCRIPTION
**Pull request by saimidu**
*Issue #, if available:*

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [x] (If applicable) I've documented below the tests I've run on the DLC image
- [x] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [x] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*
Unpin Pillow version on PT 1.7 Inference CPU/GPU and Training CPU dockerfiles to match the Training CPU dockerfile, and resolve safety check errors

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/dlc_tests/sanity/test_safety_check.py [hunk 1]
```diff
@@ -128,10 +128,10 @@ def _get_latest_package_version(package):
...
 @pytest.mark.canary("Run safety tests regularly on production images")
 @pytest.mark.skipif(not is_dlc_cicd_context(), reason="Skipping test because it is not running in dlc cicd infra")
-@pytest.mark.skipif(not (is_mainline_context() or (is_canary_context() and is_time_for_canary_safety_scan())),
-                    reason="Skipping the test to decrease the number of calls to the Safety Check DB. "
-                           "Test will be executed in the 'mainline' pipeline and canaries pipeline."
-                    )
+# @pytest.mark.skipif(not (is_mainline_context() or (is_canary_context() and is_time_for_canary_safety_scan())),
+#                     reason="Skipping the test to decrease the number of calls to the Safety Check DB. "
+#                            "Test will be executed in the 'mainline' pipeline and canaries pipeline."
+#                     )
```
Comment (saimidu) [inline test/dlc_tests/sanity/test_safety_check.py:134]: To be reverted when PR is soft-approved.

File: pytorch/buildspec.yml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
 region: &REGION <set-$REGION-in-environment>
 framework: &FRAMEWORK pytorch
-version: &VERSION 1.8.1
-short_version: &SHORT_VERSION 1.8
+version: &VERSION 1.7.1
+short_version: &SHORT_VERSION 1.7
```
Comment (saimidu) [inline pytorch/buildspec.yml:5]: To be reverted when PR is soft-approved.

**Review by jeet4320 — APPROVED** [review]
Ok to merge once tests pass

**Comment by tejaschumbalkar** [comment]
Soft Approved

**Review by mansimane — DISMISSED** [review]

File: test/dlc_tests/sanity/test_safety_check.py (hunk 1 above)
Comment (mansimane) [reply test/dlc_tests/sanity/test_safety_check.py]: Will this skip safety tests in PR?

**Comment by saimidu** [comment]
Thanks for the approvals. Merging this PR now.

**Review by mansimane — APPROVED** [review]

**Comment by saimidu** [comment]
@mansimane The pytest marker is intended to skip that test in the PR context, yes. I temporarily disabled that skip on this PR to force that test to run.

**Pull request merged**